src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilder.cs

```diff
@@ -66,11 +69,103 @@ public async IAsyncEnumerable<ProjectBuildResult> BuildProject()
         }
 
         public async Task<List<ProjectBuildResult>> Build()
+        {
+            using (var builder = new WorkspaceBuilderHelper(Logger, _workspacePath, _analyzerConfiguration))
+            {
+                var projectReferencesMap = FileUtils.GetProjectsWithReferences(_workspacePath);
+
+                if (!_analyzerConfiguration.BuildSettings.SyntaxOnly)
+                {
+                    builder.Build();
+                    foreach (var projectResult in builder.Projects)
+                    {
+                        using (ProjectBuildHandler projectBuildHandler =
+                            new ProjectBuildHandler(Logger, projectResult.Project, _analyzerConfiguration))
+                        {
+                            projectBuildHandler.AnalyzerResult = projectResult.AnalyzerResult;
+                            projectBuildHandler.ProjectAnalyzer = projectResult.ProjectAnalyzer;
+                            var result = await projectBuildHandler.Build();
```

Comment:
how expensive is this task ? can we do it parallel ? (depth 1,2,3 like this)

Reply:
This is already being done in parallel. It's invoked inside the WorkspaceBuilderHelper and has a number of configurable threads.